src/commonMain/kotlin/fr/acinq/eclair/transactions/Transactions.kt

```diff
@@ -791,17 +790,28 @@ object Transactions {
         return TransactionWithInputInfo.ClosingTx(commitTxInput, tx, toLocalOutput)
     }
 
+    private fun matchPubKeyScript(txOut: TxOut, pubkeyScript: ByteArray, amount_opt: Satoshi?): Boolean {
```

Comment:
What is the use case for  non-null `amount_opt`  ?

Reply:
It looks like a remnant from previous code, it's not used anywhere with a non-null value (not even in `eclair`).
I can remove it (it's quite brittle to rely on the amount to match anyway).

Reply:
Done in https://github.com/ACINQ/eclair-kmp/pull/236/commits/09e5ca8b4e489450631920395d40ec0c5561a2dd